embassy-rp/src/uart/mod.rs

```diff
@@ -411,6 +437,138 @@ impl<'d, T: Instance> UartRx<'d, T, Async> {
         }
         unreachable!("unrecognized rx error");
     }
+
+    /// Read from the UART, until one of the following occurs:
+    ///
+    /// * We read `buffer.len()` bytes without a line break
+    ///     * returns `Ok(buffer)`
+    /// * We read `n` bytes then a line break occurs
+    ///     * returns `Ok(&mut buffer[..n])`
+    /// * We encounter some error OTHER than a line break
+    ///     * returns `Err(Error)`
+    pub async fn read_to_break<'a>(&mut self, buffer: &'a mut [u8]) -> Result<&'a mut [u8], Error> {
```

Comment:
could you change this to return `usize` instead of a subslice? makes the signature simpler, and is more consistent with other APIs in embassy and embedded-io.

Reply:
I really don't like that signature, but I can change it back if you'd like. Will discuss in chat.